ksql-engine/src/test/java/io/confluent/ksql/ddl/commands/DropSourceCommandTest.java

```diff
@@ -15,46 +15,148 @@
 import io.confluent.ksql.metastore.StructuredDataSource;
 import io.confluent.ksql.parser.tree.DropStream;
 import io.confluent.ksql.parser.tree.QualifiedName;
-import io.confluent.ksql.services.FakeKafkaTopicClient;
+import io.confluent.ksql.serde.DataSource.DataSourceType;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConstants;
 import io.confluent.ksql.util.KsqlException;
-import io.confluent.ksql.util.MetaStoreFixture;
-import org.easymock.EasyMock;
+import java.util.Collections;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
+@RunWith(MockitoJUnitRunner.class)
 public class DropSourceCommandTest {
-  private final MetaStore metaStore = MetaStoreFixture
-      .getNewMetaStore(new InternalFunctionRegistry());
+
+  private static final String STREAM_NAME = "foo";
+  private static final String TOPIC_NAME = "foo_topic";
+
+  @Mock
+  private MetaStore metaStore;
+  @Mock
+  private KafkaTopicClient kafkaTopicClient;
+  @Mock
+  private SchemaRegistryClient schemaRegistryClient;
+  @Mock
+  private StructuredDataSource dataSource;
+  @Mock
+  private KsqlTopic ksqlTopic;
+
+  private DropSourceCommand dropSourceCommand;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setUp() {
+    when(metaStore.getSource(STREAM_NAME)).thenReturn(dataSource);
+    when(dataSource.getKsqlTopic()).thenReturn(ksqlTopic);
+    when(dataSource.getDataSourceType()).thenReturn(DataSourceType.KSTREAM);
+    when(ksqlTopic.getKafkaTopicName()).thenReturn(TOPIC_NAME);
+  }
 
   @Test
   public void shouldSucceedOnMissingSourceWithIfExists() {
-    final DropSourceCommand dropSourceCommand = new DropSourceCommand(
-        new DropStream(QualifiedName.of("foo"), true, true), StructuredDataSource.DataSourceType
-        .KSTREAM,
-        new FakeKafkaTopicClient(),
-        EasyMock.niceMock(SchemaRegistryClient.class), true);
+    // Given:
+    givenDropSourceCommand(true, true);
+    givenSourceDoesNotExist();
+
+    // When:
     final DdlCommandResult result = dropSourceCommand.run(metaStore);
+
+    // Then:
     assertThat(result.getMessage(), equalTo("Source foo does not exist."));
   }
 
   @Test
   public void shouldFailOnMissingSourceWithNoIfExists() {
-    final DropSourceCommand dropSourceCommand = new DropSourceCommand(
-        new DropStream(QualifiedName.of("foo"), false, false), StructuredDataSource.DataSourceType
-        .KSTREAM,
-        new FakeKafkaTopicClient(),
-        EasyMock.niceMock(SchemaRegistryClient.class), true);
-    try {
-      dropSourceCommand.run(metaStore);
-      fail("Should raise a Ksql Exception if source not found");
-    } catch (final KsqlException e) {}
+    // Given:
+    givenDropSourceCommand(false, true);
```

Comment:
small thing, but the tests would be more readable if you add some constants:

```suggestion
    givenDropSourceCommand(ALWAYS, WITH_DELETE_TOPIC);
    /// and
    givenDropSourceCommand(IF_EXISTS, WITHOUT_DELETE_TOPIC);
```